lib/SetAPI/generic/SetInterfaceV1.py

```diff
@@ -3,17 +3,51 @@
 
 from installed_clients.WorkspaceClient import Workspace
 
-from SetAPI.error_messages import param_required
-from SetAPI.generic.constants import ASSEMBLY, FEATURE_SET, GENOME, READS
+from SetAPI.error_messages import (
+    include_params_valid,
+    list_required,
+    no_dupes,
+    no_items,
+    param_required,
+    ref_must_be_valid,
+    ref_path_must_be_valid,
+    same_ref,
+)
+from SetAPI.generic.constants import (
+    ASSEMBLY,
+    DIFFERENTIAL_EXPRESSION_MATRIX,
+    EXPRESSION,
+    FEATURE_SET,
+    GENOME,
+    GENOME_SEARCH,
+    INC_ITEM_INFO,
+    INC_ITEM_REF_PATHS,
+    READS,
+    READS_ALIGNMENT,
+    REF_PATH_TO_SET,
+    RNASEQ_SAMPLE,
+    SAVE_SEARCH_SET,
+    SET_ITEM_NAME_TO_SET_TYPE,
+)
+from SetAPI.rnaseq_set_functions import get_rnaseq_sample_set, get_rnaseq_set
 from SetAPI.util import (
     build_ws_obj_selector,
+    check_reference,
     convert_workspace_param,
     info_to_ref,
     populate_item_object_ref_paths,
 )
 
 ALLOWS_EMPTY_SETS = [ASSEMBLY, FEATURE_SET, GENOME, READS]
 
+GENOME_REF_CHECKS = {
+    DIFFERENTIAL_EXPRESSION_MATRIX: "Genome",
+    EXPRESSION: "genome_id",
+    READS_ALIGNMENT: "genome_id",
+}
+
+ALLOWS_ZERO_GENOMES = [DIFFERENTIAL_EXPRESSION_MATRIX]
```

Comment:
a few bits and bobs here to indicate which checks need to be done on which set types